### PR TITLE
Fix for segfault on spawn failure

### PR DESCRIPTION
**Pull request by antmat**
Sometimes a segfault can happen when output in slave_t (for example on spawn error) was not initialized and timer run out.

### CODE REVIEW

**Comment by Kobolog** [comment]
What about failed-to-activate slaves? E.g., missing ld dependency: in this case we really want to see that output, but slave will never transition to active state.

**Comment by Kobolog** [comment]
@3Hren you own nodesvc, take a look.

**Comment by antmat** [comment]
Doesn't it isolate::spawn responsibility to handle such situation? For example process isolation will not raise an exception and we will be able to see output. In case of docker isolation we even can not start running application and that's why we don't have output.

File: src/service/node/slave.cpp [hunk 1]
```diff
@@ -58,6 +58,7 @@ struct slave_t::output_t  {
     asio::posix::stream_descriptor stream;
 
     output_t(unsigned long limit, std::unique_ptr<api::handle_t>&& handler, asio::io_service& loop) :
+        buffer(),
```
Comment (3Hren) [inline src/service/node/slave.cpp]: Why?

**Comment by Kobolog** [comment]
One thing is when we can't start because the binary is missing or permission issues, for example. This is an exceptional situation from the isolate p.o.v. and it can report it, but missing ld deps are undetectable, the only trace is stderr output. Same applies to app startup issues: incorrect configuration, invalid environment, etc.

**Comment by Kobolog** [comment]
And even isolates are to become responsible for such situations, I don't see it in this PR.

**Comment by 3Hren** [comment]
According to the current API, if the app cannot be spawned, we can't obtain its handle object (including access to the stdout/stderr). This is really exceptional situation, but it can be.

In the other side when it **is** spawned (i.e. we've received stdout), but we couldn't identify the app (because maybe it suddenly exits or doesn't send Handshake), we want to see its output.

**Comment by Kobolog** [comment]
Pending review: @3Hren.

**Comment by 3Hren** [comment]
Looks okay for me. Anyway, I'm going to rewrite the node/app code after c++ framework/proxy will be ready (2-4 weeks)

**Pull request merged**